language/gpt3/megatron/README.md

```diff
@@ -0,0 +1,123 @@
+# GPT-3 Reference Implementation
+
+## Setup Instructions
+
+```bash
+WORK_DIR=$PWD
+# Create Environment (conda)
+conda create -n llm python=3.9 -y
+conda activate llm
+```
+
+### Download repositories
+```bash
+cd $HOME
+git clone --recurse-submodules https://github.com/mlcommons/inference.git --depth 1
+git clone https://github.com/NVIDIA/apex.git
+git clone https://github.com/mlcommons/training.git --branch inference-megatron
+cd $HOME/apex
+git checkout -b language 2d8302a6c12e202f7b40b13a43daa95f326fd0ea
+```
+
+
+### install requirements
+```bash
+pip install torch==1.13.0 torchvision==0.14.0 datasets evaluate accelerate simplejson nltk rouge_score pybind11 Ninja numpy==1.19.5 sentencepiece
+pip install git+https://github.com/NVIDIA/mlperf-common.git
+pip install git+https://github.com/mlperf/logging.git
+
+```
+
+#### install apex
+For `pip >= 23.1`
+```bash
+cd $HOME/apex
+pip install -v --disable-pip-version-check --no-cache-dir --no-build-isolation --config-settings "--build-option=--cpp_ext" --config-settings "--build-option=--cuda_ext" ./
+```
+Otherwise
+```bash
+cd $HOME/apex
+pip install -v --disable-pip-version-check --no-cache-dir --no-build-isolation --global-option="--cpp_ext" --global-option="--cuda_ext" ./
+```
+**Warning:** Make sure the Nvidia driver version and the pytorch's version of cuda match
+
+
+### Build Loadgen
+```sh
+cd $HOME/inference/loadgen
+CFLAGS="-std=c++14 -O3" python setup.py develop
+```
+
+
+### Download & Process Dataset
+Downloads the raw data, processes and saves it as json file inside data/
+```bash
+cd $HOME/inference/language/gpt3/megatron
+python download_cnndm.py
+```
+### Calibration
+Downloads CNN-Daily Mail dataset and creates the calibration dataset (JSON) for post-training quantization
+```bash
+cd $HOME/inference/language/gpt3/megatron
+pip install datasets
+python prepare-calibration.py --calibration-list-file calibration-list.txt --output-dir </path/to/output-folder>
+```
+### Download tokenizer files
+TODO: Share tokenizer links
+
+Temporary private link:
+```bash
+cd $HOME/inference/language/gpt3/megatron/data/
+gsutil cp gs://mlperf-llm-public2/vocab/c4_en_301_5Mexp2_spm.model .
+```
+### Download GPT-3 model
+TODO: Share checkpoint link
+
+Temporary private link:
+```bash
+cd $HOME/inference/language/gpt3/megatron/
+mkdir model
+cd $HOME/inference/language/gpt3/megatron/model/
+gcloud auth login
+# gcloud storage cp "gs://mlperf-llm-public2/nv_gpt3ckpt_00011000_megatron_06162023/language_model*" .
+gsutil -m cp -r "gs://mlperf-llm-public2/nv_gpt3ckpt_00011000_megatron_06162023/language_model*" .
```

Comment:
I would suggest to use `rsync` instead of `cp`.

Reply:
Ok, but the optimizer is in that folder as well. Should we move the optimizer somewhere else?

Reply:
Does `rsync` doesn't work with wild card? In that case `cp` should be fine.